src/main/scala/com/microsoft/hyperspace/index/serde/LogicalPlanSerDeUtils.scala

```diff
@@ -135,6 +137,28 @@ object LogicalPlanSerDeUtils {
           output,
           catalogTable,
           isStreaming)
+      case LogicalRelation(
```

Comment:
Instead of this, I think you can have an extractor so that:
```scala
case LogicalRelation(
          HadoopFsRelation(
            location: InMemoryFileIndex,
            partitionSchema,
            dataSchema,
            bucketSpec,
            ExtractFileFormat(fileFormat),
            options),
          output,
          catalogTable,
          isStreaming) =>
        LogicalRelationWrapper(
          HadoopFsRelationWrapper(
            InMemoryFileIndexWrapper(location.rootPaths.map(path => path.toString)),
            partitionSchema,
            dataSchema,
            bucketSpec,
            fileFormat,
            options),
          output,
          catalogTable,
          isStreaming)
```

And `ExtractFileFormat` can create a wrapper for `CSVFileFormat` or a pass thru for others.

Reply:
added extractors